src/core/src/local_loggers/CompositeLogger.py

```diff
@@ -48,32 +51,74 @@ def log(self, message, message_type=Constants.TelemetryEventLevel.Informational)
             timestamp = self.env_layer.datetime.timestamp()
             self.file_logger.write("\n" + timestamp + "> " + message.strip(), fail_silently=False)
 
-    def log_error(self, message):
-        """log errors"""
+    def flush(self, message, message_level):
+        if message_level == Constants.MessageLevel.ERROR:
+            self.flush_error(message)
+        elif message_level == Constants.MessageLevel.WARNING:
+            self.flush_warning(message)
+        elif message_level == Constants.MessageLevel.DEBUG:
+            self.flush_debug(message)
+        elif message_level == Constants.MessageLevel.VERBOSE:
+            self.flush_verbose(message)
+        self.buffer_message = ""
+
+    def buffer_or_flush_message(self, message, buffer, message_level):
```

Comment:
What if, i keep calling log_error, log_warning etc.. with enum set to true? how will you differentiate this?

Reply:
As discussed offline, the log_error, log_warning, etc. functions were doing more preprocessing in the message string. I have renamed that functions to flush_error, flush_warning, etc. and calling them when flush is required.